### ISSUE TITLE
Dealing with Pagination or Infinite Scroll

### ISSUE THREAD

**Issue by iankrotinsky** [EatEmAll/django-djeddit#31]
Has anyone dealt with pagination or is there support somewhere in this codebase for it? Or when you get 1k posts for it to load only the first 20 and load more as you scroll?

**Comment by Logmytech** [comment]
There was a pagination app for Django which you can use. But I don't remember the name

**Comment by EatEmAll** [comment]
I haven't tested this feature in a while, but you should see "load N more comments" link once you exceeded the default limit which is currently 50. Take a look at `models.Post.getSortedReplies `, `views.loadAdditionalReplies` and at the html template thread.html:
```
        <div id="comments-callout" class="bs-callout">
            <h4 class="callout-title">{% getAmount nodes|length 'comment' %}</h4>
            {% include 'djeddit/thread_recursetree.html' %}
            {% with thread.op|missingRepliesCount:nodes as missingCount %}
                {% if missingCount %}
                    <div class="full-line additional-replies" style="margin-left:15px">
                        <br>
                        <a href="javascript:window.postFuncs.loadAdditionalReplies(
                        $('#comments-callout').find('>.additional-replies'), null, '{{ thread.op.uid }}')">
                            Load {% getAmount missingCount 'comment' infix='more' %}
                        </a>
                    </div>
                {% endif %}
            {% endwith %}
        </div>
```

Note that this link was broken until version `0.3.4`. Let me know if you have any issues making this work.

**Comment by iankrotinsky** [comment]
> I haven't tested this feature in a while, but you should see "load N more comments" link once you exceeded the default limit which is currently 50. Take a look at `models.Post.getSortedReplies `, `views.loadAdditionalReplies` and at the html template thread.html:
> 
> ```
>         <div id="comments-callout" class="bs-callout">
>             <h4 class="callout-title">{% getAmount nodes|length 'comment' %}</h4>
>             {% include 'djeddit/thread_recursetree.html' %}
>             {% with thread.op|missingRepliesCount:nodes as missingCount %}
>                 {% if missingCount %}
>                     <div class="full-line additional-replies" style="margin-left:15px">
>                         <br>
>                         <a href="javascript:window.postFuncs.loadAdditionalReplies(
>                         $('#comments-callout').find('>.additional-replies'), null, '{{ thread.op.uid }}')">
>                             Load {% getAmount missingCount 'comment' infix='more' %}
>                         </a>
>                     </div>
>                 {% endif %}
>             {% endwith %}
>         </div>
> ```
> Note that this link was broken until version `0.3.4`. Let me know if you have any issues making this work.

Is this just for comments or what about posts? If there are 1,000 posts, not comments?

**Comment by EatEmAll** [comment]
> Is this just for comments or what about posts? If there are 1,000 posts, not comments?

You're right. This feature is available only for comments, though I don't imagine that implementation of loading additional posts as you scroll down would take a lot of work. 

I'll take care of this when I'll get the chance. That's unless this is something that you can implement & contribute.

**Comment by iankrotinsky** [comment]
> > Is this just for comments or what about posts? If there are 1,000 posts, not comments?
> 
> You're right. This feature is available only for comments, though I don't imagine that implementation of loading additional posts as you scroll down would take a lot of work.
> 
> I'll take care of this when I'll get the chance. That's unless this is something that you can implement & contribute.

Unfortunately don't have the skillset to be able to implement that - if you had the time would be much appreciated.
